addons/binding/org.openhab.binding.nuki/src/main/java/org/openhab/binding/nuki/handler/NukiSmartLockHandler.java

```diff
@@ -0,0 +1,106 @@
+                .getBridgeLockState(nukiId);
+        if (bridgeLockStateResponse.getStatus() == 200) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE,
+                    "Nuki Smart Lock is " + bridgeLockStateResponse.getBridgeLockState().getStateName());
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    bridgeLockStateResponse.getMessage());
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("NukiSmartLockHandler:handleCommand({}, {})", channelUID, command);
+        String nukiId = (String) this.getConfig().get(NukiBindingConstants.CONFIG_NUKIID);
+        if (command.equals(RefreshType.REFRESH)) {
+            BridgeLockStateResponse bridgeLockStateResponse = new NukiHttpClient(getBridgeConfig())
+                    .getBridgeLockState(nukiId);
+            if (bridgeLockStateResponse.getStatus() == 200) {
+                updateState(channelUID,
+                        (bridgeLockStateResponse.getBridgeLockState().getState() == 1 ? OnOffType.ON : OnOffType.OFF));
+            } else {
+                logger.error("Could not refresh Nuki Smart Lock[{}]! Message: {}", nukiId,
+                        bridgeLockStateResponse.getMessage());
+            }
+        } else if (channelUID.getId().equals(NukiBindingConstants.CHANNEL_SMARTLOCKOPENCLOSE)
+                && (command.equals(OnOffType.ON) || command.equals(OnOffType.OFF))) {
+            int lockAction = (command.equals(OnOffType.OFF) ? NukiBindingConstants.LOCKACTIONS_UNLOCK
+                    : NukiBindingConstants.LOCKACTIONS_LOCK);
+            BridgeLockActionResponse bridgeLockActionResponse = new NukiHttpClient(getBridgeConfig())
+                    .getBridgeLockAction(nukiId, lockAction);
+            if (bridgeLockActionResponse.getStatus() != 200) {
+                logger.error("Could not execute command[{}] on Nuki Smart Lock[{}]", command, nukiId);
+            }
+        } else {
+            logger.warn("NukiSmartLockHandler:handleCommand({}, {}) not implemented!", channelUID, command);
+        }
+    }
+
+    @Override
+    public void handleUpdate(ChannelUID channelUID, State newState) {
+        logger.debug("NukiSmartLockHandler:handleUpdate({}, {})", channelUID, newState);
+        updateState(channelUID, newState);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("NukiSmartLockHandler:dispose()");
+    }
+
+    private Configuration getBridgeConfig() {
+        logger.trace("Trying to getBridgeConfig for thing[{}]", this.getThing());
+        if (this.getBridge() != null && this.getBridge().getHandler() instanceof NukiBridgeHandler) {
+            return this.getBridge().getConfiguration();
+        } else {
+            logger.error("Could not getBridgeConfig! Did you configure a bridge for this thing?");
+        }
+        return null;
```

Comment:
By returning null you will cause null pointers when checking the configuration in NukiHttpClient in the methods  `getBridgeInfo`, `getBridgeLockState` and `getBridgeLockAction`. You should not return null, but just throw an error and stop whatever the code was doing since it is not useful.